pystac/asset.py

```diff
@@ -0,0 +1,165 @@
+from copy import copy
+from typing import Any, Dict, List, Optional, TYPE_CHECKING, Union
+
+import pystac as ps
```

Comment:
I disagree. This has been super useful, and is a common pattern elsewhere. All of the types a user wants to work with are in the main package and easily accessible - no need to waste time scrolling up or checking auto-imports when you're using the core types. For other, non-core types, they exist only in their specific package and are explicitly imported.

If there's a few other folk that feel strongly about it I'd say let's open a discussion, otherwise I prefer to keep it as is.

Reply:
I personally have a hard time reading code that uses shorts for modules.  I am just not that good at reading.  I know this isn't the case for most coders

Reply:
Written on a phone, so sorry about confusions...

More specifically, I have big troubles with `ps`.  Probably extra hard for me because of the Unix `ps` command that I have used for > 30 years

Maybe going with either `pystac.Item` in the code or `from item import Item` in pystac/*.py would be good.  Then in tests, try to do things as much as possible as an external caller would?

Easier for me in the body of code:

- pystac.Item
- Item

Harder for me to read in the body:
- ps.Item

For the imports, I would prefer consistency in parts of the code, but it's not as critical.  E.g. pystac/*.py (except init), pystac/extensions/*.py, and tests/... Are 3 groups that should be the same style imports inside of the group, but maybe the groups need to be different.  

And somewhere in the tree should be a group that is as close to best practices of using pystac from outside of it.

Reply:
Meaning, I'd prefer code did `import pystac`

Reply:
My preference here would be to not use the short name internally (i.e. `import pystac` instead of `import pystac as ps`). I think it just makes the code a bit easier to read and it only adds a few characters in this case, so it seems worth it.

Reply:
Fair enough. I think the constant `pystac` takes up space where the actual types, `Collection`, `Item`, `Asset`, etc...are all STAC types that should catch the eye of the user and have a lot more meaning than the name of the library, which is why I like getting it out of the way as much as possible with `ps`. I'll likely continue that pattern into downstream work; but if there's a sense that inside this codebase it would make it better for others, I don't feel strongly enough that this is super important to hold onto, and it's a relatively quick change so I'll make it here.